src/libs/actions/Report.js

```diff
@@ -336,23 +336,14 @@ function removeOptimisticActions(reportID) {
 function updateReportWithNewAction(reportID, reportAction) {
     const newMaxSequenceNumber = reportAction.sequenceNumber;
     const isFromCurrentUser = reportAction.actorAccountID === currentUserAccountID;
-    const lastReadSequenceNumber = lastReadSequenceNumbers[reportID] || 0;
-
-    // When handling an action from the current users we can assume that their
-    // last read actionID has been updated in the server but not necessarily reflected
-    // locally so we must first update it and then calculate the unread (which should be 0)
-    if (isFromCurrentUser) {
-        setLocalLastRead(reportID, newMaxSequenceNumber);
-    }
-
+    let lastReadSequenceNumber = lastReadSequenceNumbers[reportID] || 0;
```

Comment:
I think this would be clearer if we set this up as it's own variable. So, something like 

```js
const initialLastReadSequenceNumber = lastReadSequenceNumbers[reportID] || 0;
```

Then rather than re-assign it we should do this check where it was before
```js
    if (isFromCurrentUser) {
        setLocalLastRead(reportID, newMaxSequenceNumber);
    }
```